### ISSUE TITLE
add type info to DefinitelyTyped repo

### ISSUE THREAD

**Issue by simondel** [shapesecurity/shift-parser-js#331]
Hi! Would it be possible for you to add type for `shift-parser` and `shift-codegen` to the [DefintelyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped)?

I'm one of the developers of the mutation testing framework [Stryker](https://stryker-mutator.github.io/) and I'm thinking about using `shift-parser` and `shift-codegen` instead of `esprima` and `escodegen` because of your ES6 support.

Changing our parser/codegenerator is very risky because it lies at the core of what we do: mutating code. Having type information would be very helpful to us. The [DefintelyTyped repo](https://github.com/DefinitelyTyped/DefinitelyTyped) on GitHub contains type information for modules which are not written using TypeScript. The typings of esprima may be a useful example. They can be found [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/esprima/index.d.ts).

Some places where we would use the parser, codegenerator and syntax: [parserUtils.ts](https://github.com/stryker-mutator/stryker/blob/master/src/utils/parserUtils.ts), [mutators](https://github.com/stryker-mutator/stryker/tree/master/src/mutators) and [MutatorOrchestrator.ts](https://github.com/stryker-mutator/stryker/blob/master/src/MutatorOrchestrator.ts)

**Comment by octogonz** [comment]
+1  Ideally **shift** would provide first class support for TypeScript and include .d.ts files in the `shift-parser` package.  This would avoid the need for a separate `@types/shift-parser` package (which otherwise would have to be maintained by a separate group and perhaps lag behind the official API updates).

**Comment by octogonz** [comment]
@michaelficarra are you still using **shift**?  Besides the lack of typings, the code itself doesn't declare the datatypes for any of the API functions, and I can't find any formal API docs.  Also the NPM package doesn't seem to be [very widely used](https://npm-stat.com/charts.html?package=shift-parser).  esprima seems way more popular.

**Comment by michaelficarra** [comment]
@octogonz Yes, we use the Shift tools extensively at @shapesecurity. We don't provide TypeScript types because we don't use TypeScript. You are free to define an external type declaration file, and we may even accept a PR that adds one to this repo.
